src/zenml/new/steps/step_context.py

```diff
@@ -178,18 +186,21 @@ def model(self) -> "Model":
             StepContextError: If the `Model` object is not set in `@step` or `@pipeline`.
         """
         if self.step_run.config.model is not None:
-            model = self.step_run.config.model
+            if self.step_run.model_version:
+                model = self.step_run.model_version.to_model_class()
+            else:
+                model = self.step_run.config.model
```

Comment:
This still seems a little fishy to me. We should not need the config fallback here anymore right?

Reply:
Yep, simplified